File: sunflower-core/src/main/java/org/tdf/sunflower/sync/Status.java
package org.tdf.sunflower.sync;

import org.tdf.common.util.HexBytes;


public class Status {
    private long bestBlockHeight;
    private HexBytes bestBlockHash;
    private HexBytes genesisBlockHash;
    private long prunedHeight;
    private HexBytes prunedHash;

    public Status(long bestBlockHeight, HexBytes bestBlockHash, HexBytes genesisBlockHash, long prunedHeight, HexBytes prunedHash) {
        this.bestBlockHeight = bestBlockHeight;
        this.bestBlockHash = bestBlockHash;
        this.genesisBlockHash = genesisBlockHash;
        this.prunedHeight = prunedHeight;
        this.prunedHash = prunedHash;
    }

    public Status() {
    }

    public long getBestBlockHeight() {
        return this.bestBlockHeight;
    }

    public HexBytes getBestBlockHash() {
        return this.bestBlockHash;
    }

    public HexBytes getGenesisBlockHash() {
        return this.genesisBlockHash;
    }

    public long getPrunedHeight() {
        return this.prunedHeight;
    }

    public HexBytes getPrunedHash() {
        return this.prunedHash;
    }
}

File: sunflower-core/src/main/java/org/tdf/sunflower/sync/Status.kt
package org.tdf.sunflower.sync

import com.github.salpadding.rlpstream.annotation.RlpCreator
import com.github.salpadding.rlpstream.annotation.RlpProps
import org.tdf.common.util.HexBytes

@RlpProps("bestBlockHeight", "bestBlockHash", "genesisBlockHash", "prunedHeight", "prunedHash")
data class Status @RlpCreator constructor(
    val bestBlockHeight: Long,
    val bestBlockHash: HexBytes,
    val genesisBlockHash: HexBytes,
    val prunedHeight: Long,
    val prunedHash: HexBytes,
)